### PR TITLE
Make it possible to run stemcell builder and shellout types unit tests in the same docker container or against created os image

### PR DESCRIPTION
**Pull request by StefanWutz**
**Background:**

Working on the machine id fix https://github.com/cloudfoundry/bosh-linux-stemcell-builder/pull/155 I added a new stage, but I did not added the changes to the unit tests. While my local builds and tests had been successful, the pipeline tests failed due the additional stage.

**Change**
Looking into this issue I saw that two rspec unit test groups[1] defined in this project are only executed on the ci pipeline using a trusty debootstrap environment[2] running inside a trusty docker container[3]. For the two test groups `bosh/stemcell` and `shellout_types` it seems to be an overhead as both are testing internal code and shellout types do only need a _chroot_ environment. The os image unit tests providing already this kind of environment and it is the correct os distribution to test.

This PR addresses the current problem that rspec unit tests groups (bosh/stemcell and shellout_types) can not be executed during the actual os image builds provided by this project but rather need a trusty docker image. The current unit test execution will not be changed, changes to the ci pipeline are therefore not necessary. 

In summary this PR provides the following changes:

- Removed Xenial unit tests
- Fixes an issue where changes are applied outside of the chroot instead of inside a chroot.
- Execute bosh/stemcell unit test inside the docker container: `cd /opt/bosh/bosh-stemcell; bundle exec rspec spec/`
- Execute shellout types unit tests inside the docker container: `cd /opt/bosh/bosh-stemcell; OS_IMAGE=/opt/bosh/tmp/ubuntu_base_image.tgz  bundle exec rspec spec/ --tag shellout_types`

[1] https://github.com/cloudfoundry/bosh-stemcells-ci/blob/master/tasks/test-unit.sh#L14-L17
[2] https://github.com/cloudfoundry/bosh/blob/master/ci/old-docker/main-ubuntu-chroot/Dockerfile#L5
[3] https://github.com/cloudfoundry/bosh-stemcells-ci/blob/master/tasks/test-unit.yml

### CODE REVIEW

**Comment by rkoster** [comment]
PR has been reviewed via zoom, main takeways: update PR to only include the changes made to:
- bosh-stemcell/spec/shellout_types/file_spec.rb
- bosh-stemcell/spec/shellout_types/shellout_types_helper.rb (but keep old name `spec_helper.rb`

Also check which changes where actually needed in:
- bosh-stemcell/spec/support/os_image.rb
- bosh-stemcell/spec/support/stemcell_image.rb

**Review by rkoster — CHANGES_REQUESTED** [review]
PR has been reviewed via zoom, main takeways: update PR to only include the changes made to:

bosh-stemcell/spec/shellout_types/file_spec.rb
bosh-stemcell/spec/shellout_types/shellout_types_helper.rb (but keep old name spec_helper.rb
Also check which changes where actually needed in:

bosh-stemcell/spec/support/os_image.rb
bosh-stemcell/spec/support/stemcell_image.rb

**Comment by StefanWutz** [comment]
@rkoster @ramonskie 
I have updated the PR as discussed. I tested the changes inside the _bionic_ and _main-ubuntu-chroot_ docker container.

cc: @mvach

**Comment by ystros** [comment]
Thanks @StefanWutz !
